InternshipProject#2/Manager/WatcherManager.cs

```diff
@@ -9,32 +9,55 @@ namespace InternshipProject_2.Manager;
 public class WatcherManager : IWatcherManager
 {
     private Project2Context _dbContext;
-    public HttpContext httpContext;
-
+    private readonly Mapper map;
     public WatcherManager(Project2Context dbContext)
     {
         _dbContext = dbContext;
+        map = MapperConfig.InitializeAutomapper();
     }
-    public async Task<WatchResponse> WatchTicket(WatchRequest request, int userId)
+    public async Task<WatchResponse> WatchTicket(WatchRequest request, int? userId)
     {
-        var existingWatchRequest = await _dbContext.Watchers
-        .FirstOrDefaultAsync(w => w.UserId == userId && w.TicketId == request.TicketId);
+        var isDeleted = await _dbContext.Watchers
```

Comment:
Overcomplicated.
Steps to implement:
1. find the whatcher in the db
2. if found change the db flag accordingly based on the request
3. if not found, insert a new entry if wants to watch
4. return result